dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java

```diff
@@ -1,301 +1,352 @@
-        Map<String, String> sample = url.toMap();
-        boolean result = false;
-        for (Map.Entry<String, MatchPair> matchPair : condition.entrySet()) {
-            String key = matchPair.getKey();
-            String sampleValue;
-            //get real invoked method name from invocation
-            if (invocation != null && (METHOD_KEY.equals(key) || METHODS_KEY.equals(key))) {
-                sampleValue = invocation.getMethodName();
-            } else if (ADDRESS_KEY.equals(key)) {
-                sampleValue = url.getAddress();
-            } else if (HOST_KEY.equals(key)) {
-                sampleValue = url.getHost();
-            } else {
-                sampleValue = sample.get(key);
-                if (sampleValue == null) {
-                    sampleValue = sample.get(key);
-                }
-            }
-            if (sampleValue != null) {
-                if (!matchPair.getValue().isMatch(sampleValue, param)) {
-                    return false;
-                } else {
-                    result = true;
-                }
-            } else {
-                //not pass the condition
-                if (!matchPair.getValue().matches.isEmpty()) {
-                    return false;
-                } else {
-                    result = true;
-                }
-            }
-        }
-        return result;
-    }
-
-    protected static final class MatchPair {
-        final Set<String> matches = new HashSet<String>();
-        final Set<String> mismatches = new HashSet<String>();
-
-        private boolean isMatch(String value, URL param) {
-            if (!matches.isEmpty() && mismatches.isEmpty()) {
-                for (String match : matches) {
-                    if (UrlUtils.isMatchGlobPattern(match, value, param)) {
-                        return true;
-                    }
-                }
-                return false;
-            }
-
-            if (!mismatches.isEmpty() && matches.isEmpty()) {
-                for (String mismatch : mismatches) {
-                    if (UrlUtils.isMatchGlobPattern(mismatch, value, param)) {
-                        return false;
-                    }
-                }
-                return true;
-            }
-
-            if (!matches.isEmpty() && !mismatches.isEmpty()) {
-                //when both mismatches and matches contain the same value, then using mismatches first
-                for (String mismatch : mismatches) {
-                    if (UrlUtils.isMatchGlobPattern(mismatch, value, param)) {
-                        return false;
-                    }
-                }
-                for (String match : matches) {
-                    if (UrlUtils.isMatchGlobPattern(match, value, param)) {
-                        return true;
-                    }
-                }
-                return false;
-            }
-            return false;
-        }
-    }
-}
+/*
```

Comment:
pls recover end-line format of this file

Reply:
I have fix it in https://github.com/apache/dubbo/pull/7378, please refer to the new PR.